packit_service/worker/parser.py

```diff
@@ -1133,9 +1133,19 @@ def parse_data_from_testing_farm(
             tuple: project_url, ref, result, summary, copr_build_id,
                 copr_chroot, compose, log_url, identifier
         """
-        result: TestingFarmResult = TestingFarmResult(
-            nested_get(event, "result", "overall") or event.get("state") or "unknown"
-        )
+        tf_state = event.get("state")
+        tf_result = nested_get(event, "result", "overall")
+
+        logger.debug(f"TF payload: state = {tf_state}, result['overall'] = {tf_result}")
+
+        # error and complete are the end states
+        if tf_state not in ("complete", "error"):
+            result = TestingFarmResult(tf_state)
+        else:
+            result = TestingFarmResult(tf_result or tf_state)
+
+        result = result or TestingFarmResult.unknown
```

Comment:
`TestingFarmResult.unknown` is unreachable. You either use `tf_state` or `tf_result or tf_state`, constructor will return an object and here the _unknown_ has therefore no effect.

---

Unless you can get `None` out of the constructor which, tbf, wouldn't even surprise me in this language :D